### ISSUE TITLE
[BUG] Work does not go ahead in the 'Development Environment' section.

### ISSUE THREAD

**Issue by Suzuka-Y** [KhronosGroup/Vulkan-Tutorial#53]
![90A3AF09-FB62-480B-9C90-D5889CC4A6BD_1_201_a](https://github.com/user-attachments/assets/c6023d48-6394-45ff-b36d-5c194675e248)
I'm new to Vulkan, I was going through the Vulkan tutorial 'Development Environment' and I didn't get the expected results on both WindowsOS and macOS.
![B297EE42-AAE2-4311-9CEF-3168EB297DBD](https://github.com/user-attachments/assets/1a206909-534d-4cbc-9f48-32ecf817d8bd)
macOS doesn't output the last window of this section and shows 'Build Success', but I get the error 'Thread 1: signal SIGABRT'.
![スクリーンショット 2024-11-26 011258](https://github.com/user-attachments/assets/8b696d93-e711-4b40-9ce8-20b9f0b1e987)
On WindowsOS, I'm not familiar with Visual Studio, but in the 'Let’s now configure the project to get rid of the errors. Open the project properties dialog and ensure that All Configurations is selected, because most of the settings apply to both Debug and Release mode.' section, I don't know how to display the project properties dialog, and I can't proceed from there. How can I solve this problem? Attached is a screenshot of the problem. Thank you in advance.

**Comment by SaschaWillems** [comment]
You usually can open the project properties from the context menu in the solution explorer:

![image](https://github.com/user-attachments/assets/2b6dac95-1464-4573-ba0c-f3b9113f8043)

Though I can't read your screenshots as I'm not familiar with that language, but it looks like you opened a folder instead of a proper solution, hence why you may not be able to find a project properties dialog.

**Comment by suzuka-y** [comment]
I was busy and could not reply immediately. My apologies. The Windows (Visual Studio) thing is resolved. Thank you for your help, but MacOS still stops with a 'SIGABRT' message.

**Comment by SaschaWillems** [comment]
Can you elaborate on the SIGARBT? What actual error do you get? At what point of the code does the error occur?

**Comment by suzuka-y** [comment]
When I run the program, I will see 'build success', which means that the code has been successfully built. Immediately after that, however, the program stops, as shown in the attached image.
![1305B628-3F17-44DD-A325-1A544D71B3DF](https://github.com/user-attachments/assets/5f3d6af9-b721-4039-910a-19de0f3df5da)
Perhaps the message at the bottom of the attached image indicates the cause of the error, but I do not know what the error message means.

**Comment by SaschaWillems** [comment]
That looks like some issue with code signing? I don't develop on Mac so no idea. 

@gpx1000: Any idea on what's happening here?

**Comment by gpx1000** [comment]
GLFW is the issue from the looks of it.  You're trying to load the shared library from a folder that it doesn't seem to exist in.

It also looks like the GLFW that it finds that brew installed doesn't conform to the team ID you signed your app with.  iOS requires apps be signed, I don't think that MacOS does.  You can disable signing your app in XCode and it should work.

**Comment by suzuka-y** [comment]
Thank you for your patience. We apologize for the long wait. The error has been resolved. However, the window that should appear when the program is executed does not appear. Is the cause still GLFW? Do I need to reinstall it?
![AF7D7B73-E7FD-455F-B56D-C5C318994D11](https://github.com/user-attachments/assets/2ce0040f-f4cc-46a9-919e-24c17af819cd)

**Comment by gpx1000** [comment]
Yep, GLFW handles window creation.  Whatever the bug with GLFW is, that's the cause of it given the limited information I have.

**Comment by suzuka-y** [comment]
How exactly do I disable app signing in Xcode?

**Comment by gpx1000** [comment]
this looks promising: https://stackoverflow.com/questions/31039513/how-can-i-skip-code-signing-for-development-builds-in-xcode

**Comment by suzuka-y** [comment]
When I disabled the app's signature in Xcode, I got this error. I know that it is not just a matter of emptying it. What is the next action that needs to be taken? Please let me know.
![DF6A8933-A04A-4BA9-8750-332C3F9EA888_1_201_a](https://github.com/user-attachments/assets/2a2cebed-b6f4-400e-a20d-2070fc144bb5)

**Comment by suzuka-y** [comment]
I once tried to rebuild the project from scratch. Then I realized that I had overlooked something written in the Vilkan Tutorial. It said to register the dylib file in "Frameworks and Libraries", and then add the registered file to "Copy Files" in "Build Phases" using the "+" sign. I realized that I didn't need to touch "Link Binary With Libraries" when doing this. I thought that maybe it was strange to have been dragging and dropping files into "Copy Files" and "Link Binary With Libraries" one by one. I actually followed those correct procedures to run the project, but this time I encountered an error that said "Unexpected duplicated tasks". I looked online, but couldn't find a solution.

**Comment by suzuka-y** [comment]
Solved!
As mentioned above, it was correct that I had to register the dylib file in "Frameworks and Libraries". After that, I encountered an error that said "Unexpected duplicated tasks". This was because when registering a file in "Frameworks and Libraries", an item called "Embed" is automatically added in "Build Phases", and I was able to solve the problem by manually deleting this item. Furthermore, there was also a problem with the paths registered in "Header Search Paths" and "Library Search Paths" in "Search Paths" in "Build Settings". The problem was that I had registered "opt/homebrew/Cellar/glfw/include" etc., instead of "opt/homebrew/include" or "opt/homebrew/lib", which should have been the case. Then, a window appeared when I executed the command, and I was able to achieve the desired behavior.

Final Setup:
![80BB97ED-100A-4C2A-8D39-1058BA085F08_1_201_a](https://github.com/user-attachments/assets/e523184d-95ff-4314-aaef-b80db2a5750a)
![B8344FAB-D8CA-468E-8DFD-42F4AEA5852D_1_201_a](https://github.com/user-attachments/assets/3e691145-8cfe-421b-aec3-a7dfe61b9521)
![582AC222-FFDB-4440-A6E8-F117411B5E08_1_201_a](https://github.com/user-attachments/assets/036f17d8-4d6a-47e5-8bd0-472389515a53)
![61B51D0E-6C7A-4C51-8FF7-847BFC925EA5_1_201_a](https://github.com/user-attachments/assets/6f43d7fd-81db-46ca-856d-5eadd45012ed)

**Comment by suzuka-y** [comment]
We appreciate your cooperation. Thank you very much.
